spec/lita/handlers/api/market_spec.rb

```diff
@@ -0,0 +1,188 @@
+require 'spec_helper'
+
+describe Lita::Handlers::Api::Market, lita_handler: true do
+  let(:redis) { Lita::Handlers::LunchReminder.new(robot).redis }
+  let(:karmanager) { Lita::Services::Karmanager.new(redis) }
+  let(:assigner) { Lita::Services::LunchAssigner.new(redis, karmanager) }
+  let(:market) { Lita::Services::MarketManager.new(redis, assigner, karmanager) }
+  let(:juan) { Lita::User.create(127, mention_name: 'juan') }
+  let(:pedro) { Lita::User.create(137, mention_name: 'pedro') }
+  let(:oscar) { Lita::User.create(157, mention_name: 'oscar') }
+
+  def add_limit_order(order_id, user, type, created_at)
+    order = {
+      id: order_id,
+      user_id: user.id,
+      type: type,
+      created_at: created_at
+    }.to_json
+    market.add_limit_order(order)
+    order
+  end
+
+  it { is_expected.to route_http(:get, 'market/limit_orders') }
+  it { is_expected.to route_http(:post, 'market/limit_orders') }
+  it { is_expected.to route_http(:post, 'market/market_orders') }
+
+  before do
+    ENV['MAX_LUNCHERS'] = '20'
+    assigner.add_to_lunchers('juan')
+    assigner.add_to_lunchers('pedro')
+    assigner.add_to_lunchers('oscar')
+    karmanager.set_karma(juan.id, 10)
+    karmanager.set_karma(pedro.id, 20)
+    karmanager.set_karma(oscar.id, 20)
+    assigner.add_to_current_lunchers(juan.mention_name)
+    assigner.add_to_current_lunchers(pedro.mention_name)
+    assigner.add_to_winning_lunchers(pedro.mention_name)
```

Comment:
Acá en ves de preocuparte de agregarlos con el servicio, al cambiar `assigner` por un mock solo te preocupes que cuando llames `assigner.winning_lunchers_list` te devuelva una lista de usuarios.

Esto sería cambiarlo por `allow(assigner).to receive(:winning_lunchers_list).and_return('pedro')`